assets/common.sh

```diff
@@ -117,7 +117,13 @@ log_in() {
   local username="$1"
   local password="$2"
   local registry="$3"
-  docker logout
+
+  # workaround for docker hub login problem https://github.com/concourse/docker-image-resource/pull/343 https://github.com/docker/hub-feedback/issues/1250
+  if [[ -z ${registry} ]]; then
+    # if $registry is empty, that implies that we are trying to auth to docker hub, and we need this workaround
+    docker logout https://index.docker.io/v1/
```

Comment:
Do you have a reference of the docker hub URL for log out?  How about v2?

Reply:
Alternatively, we can consider removing the 2nd `docker login` call here: https://github.com/concourse/docker-image-resource/blob/688f57658d1b187739ea407b784751a055794cc0/assets/out#L250

I think this was added in this PR https://github.com/concourse/docker-image-resource/pull/332/files

If I understand this right, we have to remove `credsStore` from `~/.docker/config.json` before running `docker build` to get the build to proceed if ECR credentials are specified.  When the build is over, we want to restore the credentials to `~/.docker/config.json`, so we run `docker login` again to do that.  It is during this 2nd docker login that we get the bad behavior for which the https://github.com/concourse/docker-image-resource/pull/343 workaround is needed.

But, wouldn't it be more expedient to run `cp ~/.docker/config.json ~/.docker/config.json.bak` before `docker build` and `mv ~/.docker/config.json.bak ~/.docker/config.json` to restore the credentials instead of running `docker login` again?

Reply:
Greetings @ericb-summit! I have a question about your PR https://github.com/concourse/docker-image-resource/pull/332.

I want to avoid calling `docker login` a 2nd time.  Do you think it would work to backup `~/.docker/config.json` prior to removing `credsStore` then restore it after `docker build` finishes?

Reply:
It should work.  I did my PR as I did to minimize the changes to the code base.  The `config.json` after logout + login should be exactly as it was before, because the way `docker login` stores creds is mind-numbingly dumb; just `base64(user:pass)` in the relevant `auths` section of `config.json`.

Reply:
> It should work. I did my PR as I did to minimize the changes to the code base. The `config.json` after logout + login should be exactly as it was before, because the way `docker login` stores creds is mind-numbingly dumb; just `base64(user:pass)` in the relevant `auths` section of `config.json`.

Thank you for confirming that for me!

Reply:
I'm marking this conversation as resolved, because the line that needed a reference has been removed.  The docker logout workaround is no longer needed, because the 2nd docker login has been removed.